source/adaptive_load/adaptive_load_controller_impl.cc

```diff
@@ -0,0 +1,444 @@
+#include <chrono>
+
+#include "envoy/common/exception.h"
+#include "envoy/config/core/v3/base.pb.h"
+
+#include "nighthawk/adaptive_load/adaptive_load_controller.h"
+#include "nighthawk/adaptive_load/metrics_plugin.h"
+#include "nighthawk/adaptive_load/scoring_function.h"
+#include "nighthawk/adaptive_load/step_controller.h"
+
+#include "external/envoy/source/common/common/logger.h"
+#include "external/envoy/source/common/common/statusor.h"
+#include "external/envoy/source/common/protobuf/protobuf.h"
+
+#include "api/adaptive_load/adaptive_load.pb.h"
+#include "api/adaptive_load/metric_spec.pb.h"
+#include "api/client/options.pb.h"
+#include "api/client/output.pb.h"
+#include "api/client/service.grpc.pb.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_join.h"
+#include "adaptive_load/metrics_plugin_impl.h"
+#include "adaptive_load/plugin_loader.h"
+
+namespace Nighthawk {
+
+namespace {
+
+using namespace Envoy; // for Envoy::Logger
```

Comment:
This is needed because the Envoy logging macros assumes the code is in namespace Envoy. I don't see a way around it.

I think most Nighthawk code is already using namespace Envoy, for the same reason, via this include file: https://github.com/envoyproxy/nighthawk/blob/6a0d3a3c9c306bf4914ddc460b6fd5931e1685e4/source/client/benchmark_client_impl.h#L34

Reply:
Yeah, this is quite annoying. Maybe we should put this on the radar by filing an issue? It would be great to resolve, to avoid the broken window effect? :-) I wonder if we can make a modification to Envoy to avoid having to do this.

Reply:
Agreed, filed #488.

Reply:
Thanks for explaining!